src/lib/capability.ts

```diff
@@ -207,8 +207,19 @@ export class Capability implements CapabilityExport {
     const bindings = this.#bindings;
     const prefix = `${this.#name}: ${model.name}`;
     const commonChain = { WithLabel, WithAnnotation, WithDeletionTimestamp, Mutate, Validate, Watch, Reconcile, Alias };
-    const isNotEmpty = (value: object) => Object.keys(value).length > 0;
-    const log = (message: string, cbString: string) => {
+
+    type CommonChainType = typeof commonChain;
+    type ExtendedCommonChainType = CommonChainType & {
+      Alias: (alias: string) => CommonChainType;
+      InNamespace: (...namespaces: string[]) => BindingWithName<T>;
+      InNamespaceRegex: (...namespaces: RegExp[]) => BindingWithName<T>;
+      WithName: (name: string) => BindingFilter<T>;
+      WithNameRegex: (regexName: RegExp) => BindingFilter<T>;
+      WithDeletionTimestamp: () => BindingFilter<T>;
```

Comment:
For `With*` We use Generics here because we have no guarantees for what's passed in as a function, right?

Looking at [`types.ts`](https://github.com/defenseunicorns/pepr/blob/main/src/lib/types.ts), I think this is an example where we can stop using `<T extends GenericClass` now that Pepr has stabilized a bit. 

I think we can specify an actual type for `In*`. Since `CommonChainType` is not exported an constrained to just `capability.ts`, I'mo pen to making that a separate issue if we feel it's appropriate.

Reply:
Yep, lets look at a second issue for this to keep focused. I like the idea but changing types would be out of scope for this.